src/cavendish_particle_tracks/_decay_angles_dialog.py

```diff
@@ -84,7 +84,8 @@ def __init__(self, parent=None):
 
     def _setup_decayangles_layer(self):
         """Create a shapes layer and add three lines to measure the Lambda, p and pi tracks"""
-
+        if "Decay Angles Tool" in self.parent.viewer.layers:
+            return self.parent.viewer.layers["Decay Angles Tool"]
```

Comment:
```suggestion

        # If layer already exists, then assume it was set up previously.
        if "Decay Angles Tool" in self.parent.viewer.layers:
            return self.parent.viewer.layers["Decay Angles Tool"]
```

Reply:
Might be worth pulling out the special, expected names of these layers and putting them as constants somewhere.
```python
# At the top of each relevant dialog maybe, or all together in a settings module. settings.py

ANGLES_LAYER_NAME = "Decay Angles Tool"
MAGNIFICATION_LAYER_NAME = "Magnification" 
...
```

... Probably don't do that in this PR.